src/plots/map/layout_attributes.js

```diff
@@ -0,0 +1,330 @@
+'use strict';
+
+var Lib = require('../../lib');
+var defaultLine = require('../../components/color').defaultLine;
+var domainAttrs = require('../domain').attributes;
+var fontAttrs = require('../font_attributes');
+var textposition = require('../../traces/scatter/attributes').textposition;
+var overrideAll = require('../../plot_api/edit_types').overrideAll;
+var templatedArray = require('../../plot_api/plot_template').templatedArray;
+
+var constants = require('./constants');
+
+var fontAttr = fontAttrs({
+    noFontVariant: true,
+    noFontShadow: true,
+    noFontLineposition: true,
+    noFontTextcase: true,
+    description: [
+        'Sets the icon text font (color=map.layer.paint.text-color, size=map.layer.layout.text-size).',
+        'Has an effect only when `type` is set to *symbol*.'
+    ].join(' ')
+});
+fontAttr.family.dflt = 'Open Sans Regular, Arial Unicode MS Regular';
+
+var attrs = module.exports = overrideAll({
+    _arrayAttrRegexps: [Lib.counterRegex('map', '.layers', true)],
+
+    domain: domainAttrs({name: 'map'}),
+
+    style: {
+        valType: 'any',
+        values: constants.styleValuesMap,
+        dflt: constants.styleValueDflt,
+        description: [
+            'Defines the map layers that are rendered by default below the trace layers defined in `data`,',
+            'which are themselves by default rendered below the layers defined in `layout.map.layers`.',
+            '',
+            'These layers can be defined either explicitly as a Map Style object which can contain multiple',
+            'layer definitions that load data from any public or private Tile Map Service (TMS or XYZ) or Web Map Service (WMS)',
+            'or implicitly by using one of the built-in style objects which use WMSes',
```

Comment:
@birkskyum I was wondering about this: "map layers that are rendered by default below the trace layers defined in `data`"

When testing the following example out on the Python side:

```

import pandas as pd
import plotly.graph_objects as go

quakes = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/earthquakes-23k.csv')


fig = go.Figure(go.Densitymap(lat=quakes.Latitude, lon=quakes.Longitude, z=quakes.Magnitude, radius=10))

fig.update_layout(
    map={
        "style": "carto-darkmatter",
        "center": {"lon": 180},
    },
    margin={"r": 0, "t": 0, "l": 0, "b": 0}
)

fig.show()
```
I see this, where it looks like the text is above the data. Is that expected.

<img width="954" alt="image" src="https://github.com/user-attachments/assets/56fcb7b1-b2d9-4396-96fb-635ab8d87579">

The previous implementation looked like this

<img width="792" alt="image" src="https://github.com/user-attachments/assets/5ff8cafe-a2aa-4815-8863-6704047134c5">

Reply:
When adding a layer to mapbox/maplibre, it's possible to specify which layer in the style it should sit below - this is thus dependent on which style is used.  In this case, with the carto dark matter style, that setting should likely be 'waterway_label' to give the expected outcome. The primary tool used to examine the individual layers of a style.json, and make new styles, is [Maputnik](https://maplibre.org/maputnik/?layer=2138096197%7E0#1.05/0/0).§

Reply:
I personally like this new look; but we should possibly try to make the maplibre version look similar to the mapbox one. @birkskyum Do you know what may change in the mock or code to make that happen?